src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -273,28 +313,75 @@ def get(self, request: Request, project: Project) -> Response:
-                            branch=current_config["config"]["defaultBranch"],
-                            path=current_config["outcome"]["sourcePath"],
+                        ref = (
+                            ctx["commit_id"]
+                            if ctx["commit_id"]
+                            else current_config["config"]["defaultBranch"]
                         )
-                        if lineCoverage and codecovUrl:
-                            result["codecov"] = {
-                                "lineCoverage": lineCoverage,
-                                "coverageUrl": codecovUrl,
-                                "status": 200,
-                            }
-
-                            if not codecov_cache:
-                                cache.set(cache_key.format(project_id=project.id), True, 3600)
+
+                        # Get commit sha from Git blame
+                        codecov_commit_sha_from_git_blame_enabled = features.has(
+                            "organizations:codecov-commit-sha-from-git-blame",
+                            project.organization,
+                            actor=request.user,
+                        )
+                        gh_integrations = integrations.filter(provider="github")
+                        should_get_commit_sha = (
+                            codecov_commit_sha_from_git_blame_enabled
+                            and len(gh_integrations) > 0
+                            and ref == current_config["config"]["defaultBranch"]
+                        )
+
+                        if should_get_commit_sha:
+                            try:
+                                integration_installation = gh_integrations[0].get_installation(
+                                    organization_id=project.organization_id
+                                )
+                                line_no = ctx.get("line_no")
+                                if line_no:
+                                    ref = self.get_latest_commit_sha_from_blame(
+                                        integration_installation,
+                                        int(line_no),
+                                        filepath,
+                                        current_config["repository"],
+                                        current_config["config"]["defaultBranch"],
+                                    )
+                            except Exception:
+                                logger.exception(
+                                    "Failed to get commit sha from git blame, pending investigation. Continuing execution."
+                                )
+
+                        if not codecov_commit_sha_from_git_blame_enabled or (
+                            codecov_commit_sha_from_git_blame_enabled
+                            and ref != current_config["config"]["defaultBranch"]
+                        ):
```

Comment:
Here is where we prevent line coverage from being shown for orgs where the `codecov_commit_sha_from_git_blame_enabled` flag is enabled and the commit sha was not successfully found from git